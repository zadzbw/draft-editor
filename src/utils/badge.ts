// 用于在浏览器控制台输出badge图标

interface BadgeParams {
  title: string;
  content: string;
}

interface LoggerParams extends BadgeParams {
  backgroundColor: string;
}

const TITLE_BGC = '#606060';
const INFO_BGC = '#1475b2';
const SUCCESS_BGC = '#42c02e';
const WARNING_BGC = '#fa0';
const ERROR_BGC = '#f50';

const logger = (...args: string[]): void => {
  // 跑测试时不进行console.log
  if (process.env.NODE_ENV === 'test') {
    return;
  }
  if (window.console && typeof window.console.log === 'function') {
    console.log(...args);
  }
};

const base = ({ title, content, backgroundColor }: LoggerParams): string[] => {
  const args = [
    `%c ${title} %c ${content} `,
    `padding: 1px; border-radius: 3px 0 0 3px; color: #fff; background: ${TITLE_BGC};`,
    `padding: 1px; border-radius: 0 3px 3px 0; color: #fff; background: ${backgroundColor};`,
  ];
  logger(...args);
  return args;
};

// 在控制台中log不同种类的badge
export function info({ title, content }: BadgeParams): string[] {
  return base({
    title,
    content,
    backgroundColor: INFO_BGC,
  });
}

export function success({ title, content }: BadgeParams): string[] {
  return base({
    title,
    content,
    backgroundColor: SUCCESS_BGC,
  });
}

export function warning({ title, content }: BadgeParams): string[] {
  return base({
    title,
    content,
    backgroundColor: WARNING_BGC,
  });
}

export function error({ title, content }: BadgeParams): string[] {
  return base({
    title,
    content,
    backgroundColor: ERROR_BGC,
  });
}
