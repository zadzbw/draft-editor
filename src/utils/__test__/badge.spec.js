import { info, success, warning, error } from '../badge';

describe('badge', () => {
  it('should log info badge correctly', () => {
    expect(info({ title: 'title1', content: 'content1' })).toMatchSnapshot();
  });

  it('should log success badge correctly', () => {
    expect(success({ title: 'title2', content: 'content2' })).toMatchSnapshot();
  });

  it('should log warning badge correctly', () => {
    expect(warning({ title: 'title3', content: 'content3' })).toMatchSnapshot();
  });

  it('should log error badge correctly', () => {
    expect(error({ title: 'title4', content: 'content4' })).toMatchSnapshot();
  });
});
