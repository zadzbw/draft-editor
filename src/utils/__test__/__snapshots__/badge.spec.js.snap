// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badge should log error badge correctly 1`] = `
Array [
  "%c title4 %c content4 ",
  "padding: 1px; border-radius: 3px 0 0 3px; color: #fff; background: #606060;",
  "padding: 1px; border-radius: 0 3px 3px 0; color: #fff; background: #f50;",
]
`;

exports[`badge should log info badge correctly 1`] = `
Array [
  "%c title1 %c content1 ",
  "padding: 1px; border-radius: 3px 0 0 3px; color: #fff; background: #606060;",
  "padding: 1px; border-radius: 0 3px 3px 0; color: #fff; background: #1475b2;",
]
`;

exports[`badge should log success badge correctly 1`] = `
Array [
  "%c title2 %c content2 ",
  "padding: 1px; border-radius: 3px 0 0 3px; color: #fff; background: #606060;",
  "padding: 1px; border-radius: 0 3px 3px 0; color: #fff; background: #42c02e;",
]
`;

exports[`badge should log warning badge correctly 1`] = `
Array [
  "%c title3 %c content3 ",
  "padding: 1px; border-radius: 3px 0 0 3px; color: #fff; background: #606060;",
  "padding: 1px; border-radius: 0 3px 3px 0; color: #fff; background: #fa0;",
]
`;
